pilot/cmd/pilot-agent/main.go

```diff
@@ -471,33 +471,11 @@ func getDNSDomain(domain string) string {
 
 // check if SDS UDS path and token path exist, if both exist, requests key/cert
 // using SDS instead of secret mount.
-func detectSds(controlPlaneBootstrap, controlPlaneAuthEnabled bool, udspath, preferJwtpath, jwtpath string) (bool, string) {
+func detectSds(udspath, preferJwtpath, jwtpath string) (bool, string) {
 	if !sdsEnabledVar.Get() {
 		return false, ""
 	}
 
-	if !controlPlaneBootstrap {
```

Comment:
I don't get it - we discussed at length that 'wait for' should only apply if it is for control plane. It seems this removed the '!controlPlane' logic - and now everything will be treated as a control plane bootstrap.

Reply:
seems controlPlaneBootstrap is always set to true and not configurable in pilot-agent https://github.com/istio/istio/blob/master/pilot/cmd/pilot-agent/main.go#L621..

Reply:
The problem is that for normal workloads ( not control plane), the logic we agreed was that we should not wait. I understand there might be a race condition or whatever you are concerned about for the control plane install, and ok with keeping the wait. 

This PR is changing the logic so that wait happens for all workloads. We don't want this - it'll slow down startup for
cases where SDS is not enabled everywhere.

Reply:
sg or workload case, for controlplane two options to move forward:
1. use a env var to indicate if sds is enabled explicitly, since we already have a ```sdsEnabledVar``` today so this PR is fine even for workload case(wait only when sds enabled). 
2. fix https://github.com/istio/istio/issues/15853 to have a usable flag in bootstrap to differentiate if sidecar is for workload or controlplane.

Reply:
So IIUC, compared with (1), (2) will support dynamic control plane SDS enablement, but it requires changes on the API and Pilot, correct?

Reply:
Or 
3. Test and possibly fix the readiness probes for pilot/mixer/galley to verify that HTTPS endpoint is valid ( has certs of some kind). If the readiness probe fails - it'll restart the container and might find the UDS path in the unlikely event of a startup race condition ( and before operator is used which would stage the install ).

Reply:
To be clear: the PR fundamental problem is that it was slowing down startup of all sidecars. 
The point of 'sds.enabled' is to allow sidecars to use SDS, if it is present. In a gradual adoption it will be enabled, so
citadel can respond to SDS requests - but we will still want to fallback. That's why we have the detection in the first place, otherwise we're back to what we had before, where setting sds is a 'global' setting. 

I don't think we need all the complexity and wait for control plane, but if you really need it and it's high priority - it can be added.

Reply:
to be clear, option 1 is already what we have today; for workload case, it only wait(and check) when ```sdsEnabledVar``` set to true, and fallback to file mount if check fail.

Reply:
Quanjie, I think Costin's point is that we just use `os.Stat(udspath)` in case it is for workload, so that we will immediately fall back to the file mount. For the Google CA case, we won't have the file mount, so that the Pilot agent will just return. Is that good enough?

Reply:
@myidpt 1. due to https://github.com/istio/istio/issues/15853, there is no way to tell in pilot-agent if it's for workload or controlplane; 2. if you look at ```waitForFile ```, it's same as using ```os.Stat(udspath)``` since ```waitForFile``` returns immediately for workload; 

The problem is without this PR, the workload case isn't handled correctly(treated as controlplane sidecar) due to https://github.com/istio/istio/issues/15853